docs/source/setup/python-advanced.rst

```diff
@@ -1,62 +1,207 @@
 Python API (advanced)
 =====================
 
-In some rare cases, experts may want to create ``Scheduler`` and ``Worker``
-objects explicitly in Python manually.  This is often necessary when making
+.. currentmodule:: distributed
+
+In some rare cases, experts may want to create ``Scheduler``, ``Worker``, and
+``Nanny``  objects explicitly in Python.  This is often necessary when making
 tools to automatically deploy Dask in custom settings.
 
-However, often it is sufficient to rely on the :doc:`Dask command line interface
-<cli>`.
+It is more common to create a :doc:`Local cluster with Client() on a single
+machine <single-distributed>` or use the :doc:`Command Line Interface (CLI) <cli>`.
+New readers are recommended to start there.
+
+If you do want to start Scheduler and Worker objects yourself you should be a
+little familiar with ``async``/``await`` style Python syntax.  These objects
+are awaitable and are commonly used within ``async with`` context managers.
+Here are a few examples to show a few ways to start and finish things.
+
+Full Example
+------------
+
+.. autosummary::
+   Scheduler
+   Worker
+   Client
+
+We first start with a comprehensive example of setting up a Scheduler, two Workers,
+and one Client in the same event loop, running a simple computation, and then
+cleaning everything up.
+
+.. code-block:: python
+
+   import asyncio
+   from dask.distributed import Scheduler, Worker, Client
+
+   async def f():
+       async with Scheduler() as s:
+           async with Worker(s.address) as w1, Worker(s.address) as w2:
+               async with Client(s.address, asynchronous=True) as client:
+                   future = client.submit(lambda x: x + 1, 10)
+                   result = await future
+                   print(result)
+
+   asyncio.get_event_loop().run_until_complete(f())
+
+Now we look at simpler examples that build up to this case.
 
 Scheduler
 ---------
 
-To start the Scheduler, provide the listening port (defaults to 8786) and Tornado
-IOLoop (defaults to ``IOLoop.current()``)
+.. autosummary::
+   Scheduler
+
+We create scheduler by creating a ``Scheduler()`` object, and then ``await``
+that object to wait for it to start up.  We can then wait on the ``.finished``
+method to wait until it closes.  In the meantime the scheduler will be active
+managing the cluster..
 
 .. code-block:: python
 
-   from distributed import Scheduler
-   from tornado.ioloop import IOLoop
-   from threading import Thread
+   import asyncio
+   from dask.distributed import Scheduler, Worker
 
-   s = Scheduler()
-   s.start('tcp://:8786')   # Listen on TCP port 8786
+   async def f():
+       s = Scheduler()        # scheduler created, but not yet running
+       s = await Scheduler()  # the scheduler is running
+       await s.finished()     # wait until the scheduler closes
 
-   loop = IOLoop.current()
-   loop.start()
+   asyncio.get_event_loop().run_until_complete(f())
```

Comment:
How can I now pass the scheduler address? Before it was possible to do `scheduler.start(address)`